src/pczt/parse.rs

```diff
@@ -0,0 +1,329 @@
+use std::collections::BTreeMap;
+
+use ff::PrimeField;
+use incrementalmerkletree::Hashable;
+use pasta_curves::pallas;
+use zcash_note_encryption::OutgoingCipherKey;
+use zip32::ChildIndex;
+
+use super::{Action, Bundle, Output, Spend, Zip32Derivation};
+use crate::{
+    bundle::Flags,
+    keys::{FullViewingKey, SpendingKey},
+    note::{ExtractedNoteCommitment, Nullifier, RandomSeed, Rho, TransmittedNoteCiphertext},
+    primitives::redpallas::{self, SpendAuth},
+    tree::{MerkleHashOrchard, MerklePath},
+    value::{NoteValue, Sign, ValueCommitTrapdoor, ValueCommitment, ValueSum},
+    Address, Anchor, Proof, NOTE_COMMITMENT_TREE_DEPTH,
+};
+
+impl Bundle {
+    /// Parses a PCZT bundle from its component parts.
+    pub fn parse(
+        actions: Vec<Action>,
+        flags: u8,
+        value_sum: (u64, bool),
```

Comment:
I decided on this serialization format for `ValueSum` (rather than e.g. `i128`) because it has no edge cases; all values are valid.